contracts/modules/token/TokenController.sol

```diff
@@ -508,6 +608,50 @@ contract TokenController is IERC1132, Iupgradable {
     emit Unlocked(_of, _reason, _amount);
   }
 
+  function withdrawCoverNote(
+    address _of,
+    bytes32[] calldata _reasons,
+    uint[] calldata _indexes
+  ) external onlyInternal {
+
+    uint reasonCount = lockReason[_of].length;
+    require(reasonCount > 0, "TokenController: no locked cover notes found");
+
+    uint lastReasonIndex = reasonCount.sub(1);
+    uint totalAmount = 0;
+
+    // The iteration is done from the last to first to prevent reason indexes from
+    // changing due to the way we delete the items (copy last to current and pop last).
+    // The provided indexes array must be ordered, otherwise reason index checks will fail.
+
+    for (uint i = _reasons.length; i > 0; i--) {
```

Comment:
find the use of i here as the "real index + 1" a bit confusing since the item you want is at i - 1.

guessing the reason for this is the underflow you would get after it hits 0